code/datums/status_effects/buffs.dm

```diff
@@ -355,6 +355,43 @@
 /datum/status_effect/regenerative_core/on_remove()
 	REMOVE_TRAIT(owner, TRAIT_IGNOREDAMAGESLOWDOWN, id)
 
+/datum/status_effect/fleshmend
+	duration = -1
+	status_type = STATUS_EFFECT_REFRESH
+	tick_interval = 1 SECONDS
+	alert_type = null
+	/// This diminishes the healing of fleshmend the higher it is.
+	var/tolerance = 1
+	var/instance_duration = 10 // in ticks
+	/// a list of integers, one for each remaining instance of fleshmend.
+	var/active_instances[0]
```

Comment:
is there any reason for this not being typecast as a list?

Reply:
Mostly because I'm not sure what best practices are for creating lists in byond coding. I'm open to suggestions.

Reply:
`var/list/active_instances` should create an empty list if that is your intention, you might need `var/list/active_instances[0]` though (this syntax is not common so im not very familiar with it)

Reply:
I'll swap that out next time I have a chance to test it and make sure that works right.

Reply:
typecasting as a list makes "active_instances[i]--" not work correctly

Reply:
Have you tried `var/list/active_instances = list()`?

Reply:
That gives me a "list index out of bound" error?

Reply:
hmm... and now I'm hitting that error even with the code unchanged, even though it'd worked before.

Reply:
Right, I see now. I think your best bet is to avoid the cursed the `var/active_instances[0]` syntax, and go with `var/list/active_instances = list()`, then rework the code for how you're handling keeping track of the instances.

Reply:
Can't believe it took me that long to figure out. I was doing the dumb thing of manipulating an array while iterating through said array. Oops

Reply:
Alright, put in a change to clean up my usage of arrays.